File: src/main/java/co/herod/adbwrapper/FixedDurationTransformer.java
package co.herod.adbwrapper;

import io.reactivex.Observable;
import io.reactivex.ObservableTransformer;

import java.util.concurrent.TimeUnit;

class FixedDurationTransformer implements ObservableTransformer<String, String> {

    private final int timeout;
    private final TimeUnit timeUnit;

    public FixedDurationTransformer(int timeout, TimeUnit timeUnit) {

        this.timeout = timeout;
        this.timeUnit = timeUnit;
    }

    @Override
    public Observable<String> apply(Observable<String> upstream) {

        return upstream
                .retry()
                .timeout(timeout, timeUnit);
    }
}

File: adb-kotlin/src/main/kotlin/co/herod/adbwrapper/rx/FixedDurationTransformer.kt
package co.herod.adbwrapper.rx

import io.reactivex.Observable
import io.reactivex.ObservableTransformer

import java.util.concurrent.TimeUnit

open class FixedDurationTransformer<T>(
        private val timeout: Int,
        private val timeUnit: TimeUnit
) : ObservableTransformer<T, T> {

    override fun apply(upstream: Observable<T>):
            Observable<T> = upstream
            // .retry()
            .timeout(timeout.toLong(), timeUnit)
}
